packages/app-bridge/src/react/useAfterInsertion.tsx

```diff
@@ -0,0 +1,39 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { useEffect } from 'react';
+
+import { type AppBridgeBlock } from '../AppBridgeBlock';
+
+/**
+ * Hook to execute a callback after the block has been inserted.
+ * The callback is only executed when the third argument is true (default).
+ *
+ * @example
+ * // Block that focuses a button on insertion after an async function has completed.
+ * const ExampleBlock = ({ appBridge }: BlockProps) => {
+ *    const buttonRef = useRef<HTMLButtonElement>(null);
+ *    const [data, setData] = useState(null);
+ *    const hasData = data !== null;
+ *
+ *    useEffect(() => {
+ *      getAsyncData().then((data) => setData(data));
+ *    }, []);
+ *
+ *    useAfterInsertion(appBridge, () => buttonRef.current?.focus(), hasData);
+ *
+ *    return hasData ?
+ *      <button ref={buttonRef} onClick={() => console.log("Creating new item...")}>Create New Item</Button> :
+ *      <div>Loading...</div>;
+ * };
+ */
+
+export const useAfterInsertion = <T extends AppBridgeBlock>(appBridge: T, callback: () => void, enabled = true) => {
```

Comment:
Why is the enabled needed? Shouldn't it be fully controlled by `isNewlyInserted`? Or is there a case I don't see?

Reply:
As described in the comment there is a case for the 'enabled'. The primary use-case thus far for this hook is so that after inserting a block you can focus the correct first element without having to click on the block again. If the block contains any asynchronous logic this would fail. 

For example, if you have a block that will call an api and only after this request has completed will display the block content, we need to be able to wait to trigger this hook